src/SWP/Bundle/AnalyticsBundle/Repository/RequestMetricRepository.php

```diff
@@ -0,0 +1,28 @@
+<?php
+
+/**
+ * This file is part of the Superdesk Web Publisher Content Bundle.
+ *
+ * Copyright 2016 Sourcefabric z.ú. and contributors.
+ *
+ * For the full copyright and license information, please see the
+ * AUTHORS and LICENSE files distributed with this source code.
+ *
+ * @copyright 2016 Sourcefabric z.ú.
+ * @license http://www.superdesk.org/license
+ */
+namespace SWP\Bundle\AnalyticsBundle\Repository;
+
+use SWP\Bundle\AnalyticsBundle\Model\RequestMetric;
+
+class RequestMetricRepository extends \Doctrine\ORM\EntityRepository
+{
+    /**
+     * @param RequestMetric $requestMetric Persists object and flushes the entity manager
+     */
+    public function persistAndFlush(RequestMetric $requestMetric)
```

Comment:
To avoid injecting the entity manager into the service, as suggested here:
http://php-and-symfony.matthiasnoback.nl/2014/05/inject-a-repository-instead-of-an-entity-manager/

On Mon, Jun 13, 2016 at 12:16 PM, Paweł Mikołajczuk <
notifications@github.com> wrote:

> In src/SWP/Bundle/AnalyticsBundle/Repository/RequestMetricRepository.php
> <https://github.com/superdesk/web-publisher/pull/78#discussion_r66768948>:
>
> > + * For the full copyright and license information, please see the
> > + * AUTHORS and LICENSE files distributed with this source code.
> > + *
> > + * @copyright 2016 Sourcefabric z.ú.
> > + * @license http://www.superdesk.org/license
> > + */
> > +namespace SWP\Bundle\AnalyticsBundle\Repository;
> > +
> > +use SWP\Bundle\AnalyticsBundle\Model\RequestMetric;
> > +
> > +class RequestMetricRepository extends \Doctrine\ORM\EntityRepository
> > +{
> > +    /**
> > +     * @param RequestMetric $requestMetric Persists object and flushes the entity manager
> > +     */
> > +    public function persistAndFlush(RequestMetric $requestMetric)
>
> Why method like that? Persisting and flushing should happen in service (or
> if there is no service yet then directly in controller). Repository should
> return Queries or Objects, never saving/persisting/modifying objects.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/superdesk/web-publisher/pull/78/files/5fbb2cfc7cad6a7bafa68838427244bcec0eb8fa#r66768948>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/ARBtFfDXeVU9SjEq7tmOZASqcRyGE0M8ks5qLS33gaJpZM4Ix7A->
> .
>